data-migrator/core/src/main/java/io/camunda/migration/data/HistoryMigrator.java

```diff
@@ -332,72 +474,181 @@ protected void migrateDecisionInstance(HistoricDecisionInstance c7DecisionInstan
-            childDecisionInstance.getDecisionDefinitionId());
-        String childDecisionInstanceId =
-            // +2 because +1 is used for the parent decision instance
-            String.format("%s-%d", c8ParentDecisionInstanceModel.decisionInstanceKey(), i + 2);
-        DecisionInstanceDbModel childDbModel = decisionInstanceConverter.apply(childDecisionInstance,
-            childDecisionInstanceId, decisionDefinition.decisionDefinitionKey(),
-            c8ParentDecisionInstanceModel.processDefinitionKey(), decisionDefinition.decisionRequirementsKey(),
-            c8ParentDecisionInstanceModel.processInstanceKey(), parentDecisionDefinitionKey,
-            c8ParentDecisionInstanceModel.flowNodeInstanceKey(), c8ParentDecisionInstanceModel.flowNodeId(),
-            dmnModelInstance);
-        c8Client.insertDecisionInstance(childDbModel);
+        try {
+          DecisionInstanceDbModel.Builder decisionInstanceDbModelBuilder = new DecisionInstanceDbModel.Builder();
+          EntityConversionContext<?, ?> context = createEntityConversionContext(childDecisionInstance,
+              HistoricDecisionInstance.class, decisionInstanceDbModelBuilder);
+          DecisionDefinitionEntity decisionDefinition = findDecisionDefinition(
+              childDecisionInstance.getDecisionDefinitionId());
+          String childDecisionInstanceId =
+              // +2 because +1 is used for the parent decision instance
+              String.format("%s-%d", c8ParentDecisionInstanceModel.decisionInstanceKey(), i + 2);
+          DmnModelInstance dmnModelInstance = c7Client.getDmnModelInstance(childDecisionInstance.getDecisionDefinitionId());
+          decisionInstanceDbModelBuilder.decisionDefinitionKey(decisionDefinition.decisionDefinitionKey())
+              .decisionInstanceId(childDecisionInstanceId)
+              .decisionRequirementsKey(decisionDefinition.decisionRequirementsKey())
+              .processDefinitionKey(c8ParentDecisionInstanceModel.processDefinitionKey())
+              .processInstanceKey(c8ParentDecisionInstanceModel.processInstanceKey())
+              .rootDecisionDefinitionKey(parentDecisionDefinitionKey)
+              .flowNodeInstanceKey(c8ParentDecisionInstanceModel.flowNodeInstanceKey())
+              .flowNodeId(c8ParentDecisionInstanceModel.flowNodeId())
+              .decisionType(determineDecisionType(dmnModelInstance, childDecisionInstance.getDecisionDefinitionKey()));
+          DecisionInstanceDbModel childDbModel = convertDecisionInstance(context);
+          c8Client.insertDecisionInstance(childDbModel);
+        } catch (EntityInterceptorException e) {
+          handleInterceptorException(childDecisionInstance.getId(), HISTORY_DECISION_INSTANCE,
+              childDecisionInstance.getEvaluationTime(), e);
+        }
       }
     }
   }
 
+  protected void processDecisionInstanceConversion(HistoricDecisionInstance c7DecisionInstance,
```

Comment:
Looking at this method and how it is chained with the `shouldSkipDecisionInstanceDueto...` helpers, I feel like the skippingDueTo checks and logic are now split between the two. Maybe the helpers are no longer needed and we could just add the checks in here? I find it confusing to read otherwise.
However, we can also refactor all this in a follow up in favour of merging quickly